server/channels/app/content_flagging_setup.go

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"sync"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/pkg/errors"
+)
+
+const contentFlaggingSetupDoneKey = "content_flagging_setup_done"
+
+var contentFlaggingGroupID string
+var contentFlaggingSetupOnce sync.Once
+
+func (a *App) GetContentFlaggingGroupID() (string, error) {
+	if contentFlaggingGroupID != "" {
+		return contentFlaggingGroupID, nil
+	}
+
+	group, err := a.Srv().propertyService.RegisterPropertyGroup(model.ContentFlaggingGroupName)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to register Content Flagging group")
+	}
+	contentFlaggingGroupID = group.ID
+
+	return contentFlaggingGroupID, nil
+}
+
+func (a *App) SetupContentFlaggingProperties() error {
+	var setupErr error
+
+	contentFlaggingSetupOnce.Do(func() {
```

Comment:
Is it really necessary to run this with `sync.Once`? Any scenarios where it might be called several times?

As far as I see this method (`SetupContentFlaggingProperties`) is being called only once with the `NewServer` function, a  call that should never called multiple times.

Also, I suggest following the pattern on similar features and handle this with `doAppMigrations()`. Or, did you thought that this is not suitable for that?

Reply:
Moved to `doAppMigrations`. No need of sync.once anymore.